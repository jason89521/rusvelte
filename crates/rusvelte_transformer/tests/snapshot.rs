#[test]
fn test() {
    insta::glob!("samples/*.svelte", |path| {
        let source = std::fs::read_to_string(path).unwrap();
        let allocator = oxc_allocator::Allocator::default();
        let mut root = rusvelte_parser::Parser::new(&source, &allocator)
            .parse()
            .expect("Parse failed");
        let (scopes, _, symbols) = rusvelte_analyzer::Analyzer::default().analyze(&root);
        let program = rusvelte_transformer::Transformer::new(&allocator, scopes, symbols)
            .transform(&mut root);
        let code = oxc_codegen::Codegen::new().build(&program).code;
        insta::assert_snapshot!("client", code)
    })
}
