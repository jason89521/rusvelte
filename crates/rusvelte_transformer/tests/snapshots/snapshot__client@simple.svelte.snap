---
source: crates/rusvelte_transformer/tests/snapshot.rs
expression: code
input_file: crates/rusvelte_transformer/tests/samples/simple.svelte
snapshot_kind: text
---
import "svelte/internal/disclose-version";
import * as $ from "svelte/internal/client";
export default function App($$anchor) {
	let count = $.state(0);
	$.next();
	var text = $.text();
	$.template_effect(() => $.set_text(text, count));
	$.append($$anchor, text);
}
